apps/desktop-wallet/src/modals/WalletRemovalModal.tsx

```diff
@@ -17,27 +17,47 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
 import { AlertTriangle } from 'lucide-react'
+import { memo } from 'react'
 import { useTranslation } from 'react-i18next'
 import { useTheme } from 'styled-components'
 
 import Button from '@/components/Button'
 import InfoBox from '@/components/InfoBox'
 import { Section } from '@/components/PageComponents/PageContainers'
 import Paragraph from '@/components/Paragraph'
+import useAnalytics from '@/features/analytics/useAnalytics'
+import { closeModal } from '@/features/modals/modalActions'
+import { ModalBaseProp } from '@/features/modals/modalTypes'
+import { useAppDispatch, useAppSelector } from '@/hooks/redux'
 import CenteredModal from '@/modals/CenteredModal'
+import { addressMetadataStorage } from '@/storage/addresses/addressMetadataPersistentStorage'
+import { activeWalletDeleted, walletDeleted } from '@/storage/wallets/walletActions'
+import { walletStorage } from '@/storage/wallets/walletPersistentStorage'
 
-interface WalletRemovalModalProps {
+export interface WalletRemovalModalProps {
+  walletId: string
   walletName: string
-  onConfirm: () => void
-  onClose: () => void
 }
 
-const WalletRemovalModal = ({ walletName, onConfirm, onClose }: WalletRemovalModalProps) => {
+const WalletRemovalModal = memo(({ id, walletId, walletName }: ModalBaseProp & WalletRemovalModalProps) => {
   const { t } = useTranslation()
   const theme = useTheme()
+  const dispatch = useAppDispatch()
+  const { sendAnalytics } = useAnalytics()
+  const activeWalletId = useAppSelector((s) => s.activeWallet.id)
+
+  const removeWallet = () => {
+    walletStorage.delete(walletId)
+    addressMetadataStorage.delete(walletId)
```

Comment:
I find it weird to not have a utility function dedicated to this.

Reply:
Also, will this throw if it fails?

Reply:
`localStorage.removeItem` doesn't throw, so no, it won't thrown. MAnaging storage is so much simpler on a browser env than on a mobile device

Reply:
Also, about the utility, I can create it with these 2 lines of code, but we are not using this code anywhere else.

Reply:
@mvaivre Do you want me to create this utility? I find it unnecessary. Lemme know

Reply:
Don't you think it could be great, just in case we want to test it at some point?
If you don't feel the need for it I'm ok with that as well.

BTW; I'm gonna approve this PR already ;)

Reply:
Added to the next release: https://github.com/alephium/alephium-frontend/issues/831